lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -120,6 +151,57 @@ TfToken GetSelectionKind()
     return TfToken();
 }
 
+// clang-format off
+TF_DEFINE_PRIVATE_TOKENS(
+    _pointInstancesPickModeTokens,
+
+    (PointInstancer)
+    (Instances)
+    (Prototypes)
+);
+// clang-format on
+
+UsdPointInstancesPickMode GetPointInstancesPickMode()
+{
+    static const MString kOptionVarName(MayaUsdOptionVars->PointInstancesPickMode.GetText());
+
+    UsdPointInstancesPickMode pickMode = UsdPointInstancesPickMode::PointInstancer;
+
+    if (MGlobal::optionVarExists(kOptionVarName)) {
+        const MString optionVarValue = MGlobal::optionVarStringValue(kOptionVarName);
+        const TfToken pickModeToken(UsdMayaUtil::convert(optionVarValue));
+
+        if (pickModeToken == _pointInstancesPickModeTokens->Instances) {
+            pickMode = UsdPointInstancesPickMode::Instances;
+        } else if (pickModeToken == _pointInstancesPickModeTokens->Prototypes) {
+            pickMode = UsdPointInstancesPickMode::Prototypes;
+        }
+    }
+
+    return pickMode;
+}
+
+//! \brief  Returns the prim or an ancestor of it that is of the given kind.
+//
+// If neither the prim itself nor any of its ancestors above it in the
+// namespace hierarchy have an authored kind that matches, an invalid null
+// prim is returned.
+UsdPrim GetPrimOrAncestorWithKind(const UsdPrim& prim, const TfToken& kind)
+{
+    UsdPrim iterPrim = prim;
+
+    while (iterPrim) {
+        TfToken primKind;
```

Comment:
If you don't mind please move the primKind declaration out of the loop.

Reply:
Addressed in 07a506398e7d60af6e5d1486c47f3a859894e272.